WordPress/Classes/Plugins/Views/InstalledPluginsListView.swift

```diff
@@ -86,13 +104,36 @@ final class InstalledPluginsListViewModel: ObservableObject {
     }
 
     func performQuery() async {
-        for await update in await self.service.installedPluginsUpdates() {
+        for await update in await self.service.installedPluginsUpdates(query: filter) {
             switch update {
             case let .success(plugins):
-                self.plugins = plugins
+                self.displayingPlugins = plugins
             case let .failure(error):
                 self.error = (error as? WpApiError)?.errorMessage ?? error.localizedDescription
             }
         }
     }
+
+    func toggle(slug: PluginSlug) async {
+        self.updating.insert(slug)
+        defer { self.updating.remove(slug) }
+
+        do {
+            try await self.service.togglePluginActivation(slug: slug)
+        } catch {
+            DDLogError("Failed to update plugin: \(error)")
```

Comment:
Are you planning to add error handling in separate PRs?

Reply:
Good catch! I forgot this part. I was planning to show a notice after the API call is completed. I'll implement it in a follow up PR.